File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/axisAPIs/AxisLayoutStackAxesVerticallyOrHorizontally.java
package com.scichart.docsandbox.examples.java.axisAPIs;

import androidx.annotation.NonNull;

import com.scichart.charting.layoutManagers.ChartLayoutState;
import com.scichart.charting.layoutManagers.DefaultLayoutManager;
import com.scichart.charting.layoutManagers.VerticalAxisLayoutStrategy;
import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.axes.AxisLayoutState;
import com.scichart.charting.visuals.axes.IAxis;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class AxisLayoutStackAxesVerticallyOrHorizontally extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) { }

    void createLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy(@NonNull SciChartSurface surface) {
        // <CreateLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
        class LeftAlignedOuterVerticallyStackedYAxisLayoutStrategy extends VerticalAxisLayoutStrategy {
            @Override
            public void measureAxes(int availableWidth, int availableHeight, ChartLayoutState chartLayoutState) {
                for (int i = 0, size = axes.size(); i < size; i++) {
                    final IAxis axis = axes.get(i);
                    axis.updateAxisMeasurements();

                    final AxisLayoutState axisLayoutState = axis.getAxisLayoutState();
                    chartLayoutState.leftOuterAreaSize = Math.max(getRequiredAxisSize(axisLayoutState), chartLayoutState.leftOuterAreaSize);
                }
            }

            @Override
            public void layoutAxes(int left, int top, int right, int bottom) {
                final int size = axes.size();
                final int height = bottom - top;
                final int axisSize = height / size;

                int topPlacement = top;
                for (int i = 0; i < size; i++) {
                    final IAxis axis = axes.get(i);
                    final AxisLayoutState axisLayoutState = axis.getAxisLayoutState();

                    final int bottomPlacement = topPlacement + axisSize;
                    axis.layoutArea(right - getRequiredAxisSize(axisLayoutState), topPlacement, right, bottomPlacement);
                    topPlacement = bottomPlacement;
                }
            }
        }
        // </CreateLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>

        // <UseLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
        // Create and set new Layout Strategy
        DefaultLayoutManager layoutManager = new DefaultLayoutManager.Builder()
                .setLeftOuterAxesLayoutStrategy(new LeftAlignedOuterVerticallyStackedYAxisLayoutStrategy())
                .build();
        surface.setLayoutManager(layoutManager);
        // </UseLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/axisAPIs/AxisLayoutStackAxesVerticallyOrHorizontally.kt
package com.scichart.docsandbox.examples.kotlin.axisAPIs

import com.scichart.charting.layoutManagers.ChartLayoutState
import com.scichart.charting.layoutManagers.DefaultLayoutManager
import com.scichart.charting.layoutManagers.VerticalAxisLayoutStrategy
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition()
class AxisLayoutStackAxesVerticallyOrHorizontally : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {}

    fun createLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy(surface: SciChartSurface) {
        // <CreateLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
        class LeftAlignedOuterVerticallyStackedYAxisLayoutStrategy :
            VerticalAxisLayoutStrategy() {
            override fun measureAxes(
                availableWidth: Int,
                availableHeight: Int,
                chartLayoutState: ChartLayoutState
            ) {
                var i = 0
                val size = axes.size
                while (i < size) {
                    val axis = axes[i]
                    axis.updateAxisMeasurements()
                    val axisLayoutState = axis.axisLayoutState
                    chartLayoutState.leftOuterAreaSize = Math.max(
                        getRequiredAxisSize(axisLayoutState),
                        chartLayoutState.leftOuterAreaSize
                    )
                    i++
                }
            }

            override fun layoutAxes(left: Int, top: Int, right: Int, bottom: Int) {
                val size = axes.size
                val height = bottom - top
                val axisSize = height / size
                var topPlacement = top
                for (i in 0 until size) {
                    val axis = axes[i]
                    val axisLayoutState = axis.axisLayoutState
                    val bottomPlacement = topPlacement + axisSize
                    axis.layoutArea(
                        right - getRequiredAxisSize(axisLayoutState),
                        topPlacement,
                        right,
                        bottomPlacement
                    )
                    topPlacement = bottomPlacement
                }
            }
        }
        // </CreateLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>

        // <UseLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
        // Create and set new Layout Strategy
        val layoutManager = DefaultLayoutManager.Builder()
            .setLeftOuterAxesLayoutStrategy(LeftAlignedOuterVerticallyStackedYAxisLayoutStrategy())
            .build()
        surface.layoutManager = layoutManager
        // </UseLeftAlignedOuterVerticallyStackedYAxisLayoutStrategy>
    }
}
